### PR TITLE
Example setup for a Node http server w/test

### PR DESCRIPTION
**Pull request by regiellis**
Basic setup for a node core server, should work with any framework based on node.

### CODE REVIEW

File: node/www.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+const server = http.createServer(internal.RequestHandler);
+
+process.on('uncaughtException', (err) => {
+    process.stderr.write(`Something blew up: ${err.message} \n`);
+    process.stderr.write(`Trace the explosion from here: \n ${err.stack}`); 
+    process.exit(1);
+});
+
+module.exports = server;
```
Comment (mholt) [inline node/www.js]: I am not sure we need a Node program in this repo. It goes outside the scope of this project.

File: node/spec-test.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+    describe('/', () => {
+        it('should return a JSON template', (done) => {
+            http.get(internal.options, internal.reply);
+            done(); 
+        });
+    });
+});
+
+
```
Comment (mholt) [inline node/spec-test.js]: Same here, let's get rid of all the Node code.

File: node/server.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    if (err) return process.stdout.write(`Something went wrong: ${err} \n`);
+    process.stdout.write(`Node Server running on port ${port} \n`);
+});
+
+process.on('uncaughtException', (err) => {
+    process.stderr.write(`Something blew up: ${err.message} \n`);
+    process.stderr.write(`Trace the explosion from here: \n ${err.stack}`); 
+    process.exit(1);
+});
+
```
Comment (mholt) [inline node/server.js]: And here. I'm sure people can go somewhere to learn how to write a simple server in Node, but coming to a Caddy examples repo is a bit of a stretch.

File: node/package.json [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  "scripts": {
+    "start": "node server.js",
+    "test": "mocha -R spec spec-test.js"
+  },
+  "license": "MIT",
+  "devDependencies": {
+    "chai": "^3.5.0",
+    "mocha": "^3.1.0"
+  }
+}
```
Comment (mholt) [inline node/package.json]: Which means that we can drop this file too.

File: node/caddy.json [hunk 5]
```diff
@@ -0,0 +1,4 @@
+{
+  "status": 200,
+  "msg": "Caddy is serving just fine"
+}
```
Comment (mholt) [inline node/caddy.json]: And we probably don't need this either. :)

**Comment by regiellis** [comment]
Hey Matt, 
I get where you are coming from; put this together as a example for my team in my company, just thought I submit it. I am not sure if you want me to update the example or cancel the pull request based on your first and the last comment.

**Comment by mholt** [comment]
I think this would make a fine blog post. :+1: But this example is so Node.js-heavy that I don't think it belongs here, so I won't be accepting this as-is. And then after we strip it down, what's left is literally 2 lines of Caddyfile. And that's actually great -- it shows in some sense how easy Caddy is to use...

Perhaps to make it more proper for this repo, submit a README along the Caddyfile and describe how to set up a Node backend (you can include some Node code blocks even, but maybe don't need a whole Node project!) and explain that all you have to do is reverse-proxy to it, since that's really all that Caddy does in this case, is a very vanilla reverse proxy... it has nothing to do with Node specifically, but maybe it will draw the attention of some Google traffic. :)

**Comment by regiellis** [comment]
Sounds good Matt, I will turn it in to blog post on Medium this week. Let you know when it goes up. Since I have started my deep dive in to the golang, I will make my attempts to help there.

**Comment by mholt** [comment]
Sounds good! If you still want to submit this PR (modified), feel free to do so, otherwise you can close it. :+1:

**Pull request closed without merging**